sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -622,7 +622,7 @@ case class GpuCast(
     def throwIfAny(cv: ColumnView): Unit = {
       withResource(cv) { cv =>
         withResource(cv.any()) { isAny =>
-          if (isAny.getBoolean) {
+          if (isAny.isValid && isAny.getBoolean) {
```

Comment:
Are there any tests that hit this, or any of the others?  All nulls is not that common, but I am a bit concerned about some of the cast logic and I am not sure if all nulls in an ANSI cast is doing exactly what we want without a test for it.

Reply:
@andygrove did add tests for nulls which found some issues back when he fixed some `any`/`all` usage in #2933.  I'll try to make a test for this.  Note that `any`/`all` will also return a null scalar if the input column is empty in case this ever gets invoked on a degenerate batch.

Reply:
I added some tests that exercise these code paths, and I used a custom cudf that asserts in the Scalar.get* methods to verify they fail without the changes and pass afterwards.